### PR TITLE
[gclient] Get rid of GearApiWithNode

### LINKED ISSUE #2397: gclient: `GearApiWithNode` doesn't support login operation
### File Location(s)

gclient/src/api/mod.rs

### Proposal

`GearApiWithNode` can't provide `GearApi::with` method as it implements `Deref` that can use `&self` only instead of `self`.
Also, it is quite inconvenient to have two structures (`GearApiWithNode` and `GearApi`).

Consider to combine the node functionality in `GearApi`.

### PR DESCRIPTION
**Pull request by DennisInSky**
Resolves #2397  .

`Node` has been moved insied `GearApi` as an option

@reviewer-or-team

### CODE REVIEW

File: gclient/src/node/mod.rs [hunk 1]
```diff
@@ -25,44 +25,23 @@ use std::{
...
-        )
-    }
-
     /// Returns Web Socket address the node is listening to.
     pub fn ws_address(&self) -> &WSAddress {
         &self.ws_address
     }
 
-    fn try_node_from_path(path: impl AsRef<OsStr>, args: Vec<&str>) -> Result<Self> {
+    pub fn try_node_from_path(path: impl AsRef<OsStr>, args: Vec<&str>) -> Result<Self> {
```
Comment (shamilsan) [inline gclient/src/node/mod.rs:44]: Maybe just `try_from_path`?

File: gclient/src/api/mod.rs [hunk 2]
```diff
@@ -28,11 +28,11 @@ use crate::{
 };
 use error::*;
 use gsdk::{ext::sp_runtime::AccountId32, signer::Signer, Api};
-use std::{marker::PhantomData, ops::Deref};
+use std::{ffi::OsStr, sync::Arc};
 
 /// The API instance contains methods to access the node.
 #[derive(Clone)]
-pub struct GearApi(Signer);
+pub struct GearApi(Signer, Option<Arc<Node>>);
```
Comment (shamilsan) [inline gclient/src/api/mod.rs:35]: I'd change it to struct
```suggestion
pub struct GearApi {
    signer: Signer, 
    node: Option<Arc<Node>>
}
```

**Review by shamilsan — APPROVED** [review]
I like it 👍

File: gclient/src/api/mod.rs (hunk 2 above)
Comment (DennisInSky) [reply gclient/src/api/mod.rs:35]: @shamilsan are you sure about this. I have done it locally, but it imposes a lot of changes.

Comment (shamilsan) [reply gclient/src/api/mod.rs:35]: As it doesn't affect external interface (only internal implementation), we can do it later, ok.

**Pull request merged**